homeassistant/helpers/helper_config_entry_flow.py

```diff
@@ -66,6 +81,25 @@ def __init_subclass__(cls, **kwargs: Any) -> None:
         """Initialize a subclass, register if possible."""
         super().__init_subclass__(**kwargs)
 
+        @callback
+        def _async_get_options_flow(
+            config_entry: config_entries.ConfigEntry,
+        ) -> config_entries.OptionsFlow:
+            """Get the options flow for this handler."""
+            return HelperOptionsFlowHandler(
+                config_entry,
+                cls.steps,
+                cls.async_config_entry_title,
+                cls.async_initial_options_step,
+                cls.async_next_step,
+                cls.async_validate_input,
+            )
+
+        if (
+            cls.async_initial_options_step
+            is not HelperConfigFlowHandler.async_initial_options_step
+        ):
+            cls.async_get_options_flow = _async_get_options_flow  # type: ignore[assignment]
```

Comment:
why don't we put this if-check inside the `async_get_options_flow`?

Reply:
Because that breaks `ConfigFlow.async_supports_options_flow`, updated so we overload `async_supports_options_flow` instead, maybe that's a bit clearer.